### PR TITLE
add doc for master config

### PR DESCRIPTION
**Pull request by tsaridas**
### What does this PR do?
adds documentation for https://github.com/saltstack/salt/pull/38086

@rallytime I wouldn't know what to add to Release notes. 
Please advise.

### CODE REVIEW

File: doc/ref/configuration/master.rst [hunk 1]
```diff
@@ -3121,6 +3121,23 @@ order will be used.
 
     syndic_failover: random
 
+. conf_master:: syndic_forward_all_events
```
Comment (rallytime) [inline doc/ref/configuration/master.rst]: Can you add an additional `.` here at the beginning of the line?

File: doc/ref/configuration/master.rst [hunk 2]
```diff
@@ -3121,6 +3121,23 @@ order will be used.
...
+
+``syndic_forward_all_events`
```
Comment (rallytime) [inline doc/ref/configuration/master.rst]: Can you add a final "`" to the end of the line?

File: doc/ref/configuration/master.rst [hunk 3]
```diff
@@ -3121,6 +3121,23 @@ order will be used.
...
+-------------------
+
+.. versionadded:: Nitrogen
+
+Default: ``False``
+
+Option on multi-syndic or single when connected to multiple masters to be able to
+sent events to all connected masters.
```
Comment (rallytime) [inline doc/ref/configuration/master.rst]: Can you fix up this small typo, too? This should be ``send`` instead of ``sent``.

**Review by rallytime — CHANGES_REQUESTED** [review]
Awesome! Thanks @tsaridas for adding these docs. It is very much appreciated. 

I had a couple of small requests that I commented about. For the release notes, I think maybe adding a new section header for master config additions with this option and a short description would be perfect. Something like:
```
Master Configuration Additions
==============================

- ``syndic_forward_all_events``: Option on multi-syndic or single when connected
  to multiple masters to be able to send events to all connected masters.
```
Or something along those lines. Then we can get this in. Thanks again!

**Comment by tsaridas** [comment]
done, thanks for the comments

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
Thanks again!

**Pull request merged**